### PR TITLE
Release 21 rc1

### LINKED ISSUE #3786: [Tracking] v0.21.0RC1 release
### Release Core
- [x] [Engineering] Verify all necessary changes exist on the release branch
- [x] [Engineering] Follow [dbt-release workflow](https://www.notion.so/dbtlabs/Releasing-b97c5ea9a02949e79e81db3566bbc8ef#03ff37da697d4d8ba63d24fae1bfa817) 
- [x] [Product] Update migration guide (next.docs.getdbt.com)

### Release Cloud
- [ ] [Engineering] Create a platform issue to update dbt Cloud and verify it is completed. [Example issue](https://github.com/dbt-labs/dbt-cloud/issues/3481)
- [ ] [Engineering] Determine if schemas have changed. If so, generate new schemas and push to schemas.getdbt.com

### Announce
- [x] [Product] Publish discourse
- [x] [Product] Announce in dbt Slack

### Post-release
- [x] [Engineering] [Bump plugin versions](https://www.notion.so/dbtlabs/Releasing-b97c5ea9a02949e79e81db3566bbc8ef#f01854e8da3641179fbcbe505bdf515c) (dbt-spark + dbt-presto), add compatibility as needed
   - [x]  [Spark](https://github.com/dbt-labs/dbt-spark) 
   - [x]  [Presto](https://github.com/dbt-labs/dbt-presto)
- [ ] [Engineering] Create a platform issue to update dbt-spark versions to dbt Cloud. [Example issue](https://github.com/dbt-labs/dbt-cloud/issues/3481)
- [x] [Product] Release new version of dbt-utils with new dbt version compatibility. If there are breaking changes requiring a minor version, plan upgrades of other packages that depend on dbt-utils.
- [ ] [Engineering] Create an epic for the final release

### PR DESCRIPTION
**Pull request by gshank**
resolves #3786

### Description

Branch to switch 0.21.latest to develop

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,57 @@
 ## dbt 0.21.0 (Release TBD)
 
+## dbt 0.21.RC1 (September 17, 2021)
```
Comment (leahwicz) [inline CHANGELOG.md:3]: The release process auto-updates this so you don't have to do it here. If we just remove this line then the GitHub Action will do it for us

Comment (leahwicz) [inline CHANGELOG.md:3]:
```suggestion
```

**Review by leahwicz — APPROVED** [review]
Other than the Changelog comment, LGTM

**Comment by leahwicz** [comment]
@cla-bot check

**Comment by jtcohen6** [comment]
e634420 is associated with Joel's old work email, which he's (sensibly) removed from his GitHub account (which has signed the CLA). Because of that, the cla-bot isn't able to associate the commit with a CLA-compliant account. I think we may need to rebase this PR and edit the author for that commit, as I did for 0070cd9.

**Review by kwigley — APPROVED** [review]

**Pull request merged**